src/java.desktop/windows/native/libawt/windows/awt_Component.cpp

```diff
@@ -3617,7 +3617,7 @@ UINT AwtComponent::WindowsKeyToJavaChar(UINT wkey, UINT modifiers, TransOps ops,
     WORD wChar[2];
     int converted = 1;
     UINT ch = ::MapVirtualKeyEx(wkey, 2, GetKeyboardLayout());
-    if (ch & 0x80000000) {
+    if (ch & 0x80000000 && ch != 0x800000A8) {
```

Comment:
1. It doesn't work for third key press in case of key's pressing 3 times in a row
2. Adding the workaround here is a quite dangerous because it affects the state of the `deadKeyActive` flag. I think it could break some exotic cases, e.g. when the char is used as a [chained dead key](https://en.wikipedia.org/wiki/Dead_key#Chained_dead_keys).

Actually I don't believe that adjusting logic of reporting keyChars for KEY_PRESSED events is the right way. The specification clearly says
> the result of the getKeyChar method is guaranteed to be meaningful only for KEY_TYPED events

So clients can't depend on that IMHO.